.eslintrc.js

```diff
@@ -1,15 +1,24 @@
 module.exports = {
   root: true,
   extends: './node_modules/@hashicorp/platform-cli/config/.eslintrc.js',
+
   rules: {
     curly: 'error',
-    '@typescript-eslint/explicit-module-boundary-types': 'off',
-    '@typescript-eslint/typedef': [
-      'warn',
-      {
-        arrowParameter: true,
-        parameter: true,
-      },
-    ],
   },
+
+  overrides: [
+    {
+      files: ['**/*.ts?(x)'],
+      rules: {
+        '@typescript-eslint/explicit-module-boundary-types': 'off',
+        '@typescript-eslint/typedef': [
+          'warn',
+          {
+            arrowParameter: true,
+            parameter: true,
+          },
+        ],
```

Comment:
I shifted these TS-specific rules to an overrides block. This ensures that the rules aren't applied to `.js` files.